File: apps/mobile/android/app/src/main/java/com/tchat/screens/SocialScreen.kt
package com.tchat.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tchat.designsystem.Colors
import com.tchat.designsystem.Spacing

/**
 * Social media feed interface screen
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SocialScreen() {
    var selectedTab by remember { mutableStateOf(SocialTab.FEED) }
    var showingCamera by remember { mutableStateOf(false) }

    // Mock posts
    val posts = listOf(
        Post("Alice Johnson", "2h", "Just finished my morning workout! 💪 #fitness", Icons.Default.DirectionsWalk, 23, 5),
        Post("Bob Smith", "4h", "Beautiful sunset from my balcony 🌅", Icons.Default.WbSunny, 67, 12),
        Post("Carol Davis", "6h", "New coffee shop opened downtown! ☕️ Must try", Icons.Default.Coffee, 45, 8),
        Post("David Wilson", "8h", "Working on a new project. Excited to share soon! 🚀", Icons.Default.Laptop, 89, 15),
        Post("Emma Brown", "12h", "Weekend hiking adventure! Nature is amazing 🏔️", Icons.Default.Terrain, 156, 28)
    )

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Colors.background)
    ) {
        // Top app bar
        TopAppBar(
            title = {
                Text(
                    text = "Social",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Colors.textPrimary
                )
            },
            actions = {
                IconButton(onClick = { showingCamera = true }) {
                    Icon(
                        imageVector = Icons.Default.CameraAlt,
                        contentDescription = "Camera",
                        tint = Colors.primary
                    )
                }
            },
            colors = TopAppBarDefaults.topAppBarColors(
                containerColor = Colors.background
            )
        )

        // Tab selector
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(Colors.surface)
        ) {
            SocialTab.values().forEach { tab ->
                Column(
                    modifier = Modifier
                        .weight(1f)
                        .clickable { selectedTab = tab }
                        .padding(vertical = Spacing.sm),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        imageVector = tab.icon,
                        contentDescription = tab.title,
                        tint = if (selectedTab == tab) Colors.primary else Colors.textSecondary,
                        modifier = Modifier.size(16.dp)
                    )
                    Spacer(modifier = Modifier.height(Spacing.xs))
                    Text(
                        text = tab.title,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Medium,
                        color = if (selectedTab == tab) Colors.primary else Colors.textSecondary
                    )
                }
            }
        }

        // Content based on selected tab
        when (selectedTab) {
            SocialTab.FEED -> FeedView(posts = posts)
            SocialTab.DISCOVER -> DiscoverView()
            SocialTab.NOTIFICATIONS -> NotificationsView()
        }
    }
}

// MARK: - Social Tab Enum
enum class SocialTab(
    val title: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
) {
    FEED("Feed", Icons.Default.Home),
    DISCOVER("Discover", Icons.Default.Explore),
    NOTIFICATIONS("Alerts", Icons.Default.Notifications)
}

// MARK: - Data Classes
data class Post(
    val author: String,
    val time: String,
    val content: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector,
    val likes: Int,
    val comments: Int
)

// MARK: - Feed View
@Composable
private fun FeedView(posts: List<Post>) {
    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = PaddingValues(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.md)
    ) {
        items(posts) { post ->
            PostCard(post = post)
        }
    }
}

// MARK: - Post Card Component
@Composable
private fun PostCard(post: Post) {
    var isLiked by remember { mutableStateOf(false) }
    var currentLikes by remember { mutableStateOf(post.likes) }

    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = androidx.compose.ui.graphics.Color.White
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Column(
            modifier = Modifier.padding(Spacing.md)
        ) {
            // Header
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .background(
                            color = Colors.primary.copy(alpha = 0.2f),
                            shape = CircleShape
                        ),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = post.author.first().toString(),
                        fontSize = 16.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Colors.primary
                    )
                }

                Spacer(modifier = Modifier.width(Spacing.md))

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = post.author,
                        fontSize = 14.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Colors.textPrimary
                    )
                    Text(
                        text = post.time,
                        fontSize = 12.sp,
                        color = Colors.textSecondary
                    )
                }

                IconButton(onClick = { /* More options */ }) {
                    Icon(
                        imageVector = Icons.Default.MoreVert,
                        contentDescription = "More",
                        tint = Colors.textSecondary
                    )
                }
            }

            Spacer(modifier = Modifier.height(Spacing.sm))

            // Content
            Text(
                text = post.content,
                fontSize = 15.sp,
                color = Colors.textPrimary
            )

            Spacer(modifier = Modifier.height(Spacing.sm))

            // Icon/Media placeholder
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(80.dp)
                    .background(
                        color = Colors.surface,
                        shape = RoundedCornerShape(12.dp)
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = post.icon,
                    contentDescription = "Content",
                    tint = Colors.primary,
                    modifier = Modifier.size(24.dp)
                )
            }

            Spacer(modifier = Modifier.height(Spacing.sm))

            // Actions
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(Spacing.lg)
            ) {
                Row(
                    modifier = Modifier.clickable {
                        isLiked = !isLiked
                        currentLikes += if (isLiked) 1 else -1
                    },
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = if (isLiked) Icons.Default.Favorite else Icons.Default.FavoriteBorder,
                        contentDescription = "Like",
                        tint = if (isLiked) androidx.compose.ui.graphics.Color.Red else Colors.textSecondary,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(modifier = Modifier.width(Spacing.xs))
                    Text(
                        text = currentLikes.toString(),
                        fontSize = 14.sp,
                        color = Colors.textSecondary
                    )
                }

                Row(
                    modifier = Modifier.clickable { /* Comment */ },
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.ChatBubbleOutline,
                        contentDescription = "Comment",
                        tint = Colors.textSecondary,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(modifier = Modifier.width(Spacing.xs))
                    Text(
                        text = post.comments.toString(),
                        fontSize = 14.sp,
                        color = Colors.textSecondary
                    )
                }

                IconButton(onClick = { /* Share */ }) {
                    Icon(
                        imageVector = Icons.Default.Share,
                        contentDescription = "Share",
                        tint = Colors.textSecondary,
                        modifier = Modifier.size(20.dp)
                    )
                }
            }
        }
    }
}

// MARK: - Discover View
@Composable
private fun DiscoverView() {
    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = PaddingValues(Spacing.md)
    ) {
        item {
            Text(
                text = "Trending Topics",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Colors.textPrimary,
                modifier = Modifier.padding(bottom = Spacing.md)
            )
        }

        item {
            LazyVerticalGrid(
                columns = GridCells.Fixed(2),
                modifier = Modifier.height(360.dp),
                horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                verticalArrangement = Arrangement.spacedBy(Spacing.sm)
            ) {
                items(6) { index ->
                    Card(
                        modifier = Modifier.height(120.dp),
                        shape = RoundedCornerShape(12.dp),
                        colors = CardDefaults.cardColors(
                            containerColor = Colors.surface
                        )
                    ) {
                        Column(
                            modifier = Modifier.fillMaxSize(),
                            horizontalAlignment = Alignment.CenterHorizontally,
                            verticalArrangement = Arrangement.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Whatshot,
                                contentDescription = "Trending",
                                tint = Colors.primary,
                                modifier = Modifier.size(24.dp)
                            )
                            Spacer(modifier = Modifier.height(Spacing.xs))
                            Text(
                                text = "Topic ${index + 1}",
                                fontSize = 14.sp,
                                fontWeight = FontWeight.Medium,
                                color = Colors.textPrimary
                            )
                        }
                    }
                }
            }
        }
    }
}

// MARK: - Notifications View
@Composable
private fun NotificationsView() {
    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = PaddingValues(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.sm)
    ) {
        items(5) { index ->
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = Spacing.md, vertical = Spacing.sm),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .background(
                            color = Colors.primary.copy(alpha = 0.2f),
                            shape = CircleShape
                        ),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.Notifications,
                        contentDescription = "Notification",
                        tint = Colors.primary,
                        modifier = Modifier.size(16.dp)
                    )
                }

                Spacer(modifier = Modifier.width(Spacing.md))

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = "Notification ${index + 1}",
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Medium,
                        color = Colors.textPrimary
                    )
                    Text(
                        text = "This is a sample notification message",
                        fontSize = 12.sp,
                        color = Colors.textSecondary
                    )
                }

                Text(
                    text = "2m",
                    fontSize = 12.sp,
                    color = Colors.textSecondary
                )
            }
        }
    }
}

// MARK: - Preview
@Preview(showBackground = true)
@Composable
fun SocialScreenPreview() {
    SocialScreen()
}

File: apps/mobile/ios/Sources/Screens/SocialScreen.swift
//
//  SocialScreen.swift
//  TchatApp
//
//  Created by Claude on 22/09/2024.
//

import SwiftUI

/// Social media feed interface screen
public struct SocialScreen: View {
    @State private var selectedTab: SocialTab = .feed
    @State private var showingCamera = false

    private let colors = Colors()
    private let spacing = Spacing()

    // Mock posts
    private let posts = [
        ("Alice Johnson", "2h", "Just finished my morning workout! 💪 #fitness", "figure.walk", 23, 5),
        ("Bob Smith", "4h", "Beautiful sunset from my balcony 🌅", "sun.max.fill", 67, 12),
        ("Carol Davis", "6h", "New coffee shop opened downtown! ☕️ Must try", "cup.and.saucer.fill", 45, 8),
        ("David Wilson", "8h", "Working on a new project. Excited to share soon! 🚀", "laptop", 89, 15),
        ("Emma Brown", "12h", "Weekend hiking adventure! Nature is amazing 🏔️", "mountain.2.fill", 156, 28)
    ]

    public init() {}

    public var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Segmented control
                HStack(spacing: 0) {
                    ForEach(SocialTab.allCases, id: \.self) { tab in
                        Button(action: {
                            selectedTab = tab
                        }) {
                            VStack(spacing: spacing.xs) {
                                Image(systemName: tab.icon)
                                    .font(.system(size: 16))
                                Text(tab.title)
                                    .font(.system(size: 12, weight: .medium))
                            }
                            .foregroundColor(selectedTab == tab ? colors.primary : colors.textSecondary)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, spacing.sm)
                        }
                    }
                }
                .background(colors.surface)

                // Content based on selected tab
                switch selectedTab {
                case .feed:
                    FeedView(posts: posts)
                case .discover:
                    DiscoverView()
                case .notifications:
                    NotificationsView()
                }
            }
            .navigationTitle("Social")
            .navigationBarTitleDisplayMode(.large)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: {
                        showingCamera = true
                    }) {
                        Image(systemName: "camera.fill")
                            .foregroundColor(colors.primary)
                    }
                }
            }
        }
        .navigationViewStyle(StackNavigationViewStyle())
        .sheet(isPresented: $showingCamera) {
            CameraView()
        }
    }
}

// MARK: - Social Tab Enum
enum SocialTab: CaseIterable {
    case feed, discover, notifications

    var title: String {
        switch self {
        case .feed: return "Feed"
        case .discover: return "Discover"
        case .notifications: return "Alerts"
        }
    }

    var icon: String {
        switch self {
        case .feed: return "house.fill"
        case .discover: return "safari.fill"
        case .notifications: return "bell.fill"
        }
    }
}

// MARK: - Feed View
private struct FeedView: View {
    let posts: [(String, String, String, String, Int, Int)]

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        ScrollView {
            LazyVStack(spacing: spacing.md) {
                ForEach(Array(posts.enumerated()), id: \.offset) { index, post in
                    PostCard(
                        author: post.0,
                        time: post.1,
                        content: post.2,
                        icon: post.3,
                        likes: post.4,
                        comments: post.5
                    )
                }
            }
            .padding(.horizontal, spacing.md)
            .padding(.top, spacing.sm)
        }
    }
}

// MARK: - Post Card Component
private struct PostCard: View {
    let author: String
    let time: String
    let content: String
    let icon: String
    let likes: Int
    let comments: Int

    @State private var isLiked = false
    @State private var currentLikes: Int

    private let colors = Colors()
    private let spacing = Spacing()

    init(author: String, time: String, content: String, icon: String, likes: Int, comments: Int) {
        self.author = author
        self.time = time
        self.content = content
        self.icon = icon
        self.likes = likes
        self.comments = comments
        self._currentLikes = State(initialValue: likes)
    }

    var body: some View {
        VStack(alignment: .leading, spacing: spacing.sm) {
            // Header
            HStack {
                Circle()
                    .fill(colors.primary.opacity(0.2))
                    .frame(width: 40, height: 40)
                    .overlay(
                        Text(author.prefix(1))
                            .font(.system(size: 16, weight: .semibold))
                            .foregroundColor(colors.primary)
                    )

                VStack(alignment: .leading, spacing: 2) {
                    Text(author)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(colors.textPrimary)

                    Text(time)
                        .font(.system(size: 12))
                        .foregroundColor(colors.textSecondary)
                }

                Spacer()

                Button(action: {}) {
                    Image(systemName: "ellipsis")
                        .foregroundColor(colors.textSecondary)
                }
            }

            // Content
            Text(content)
                .font(.system(size: 15))
                .foregroundColor(colors.textPrimary)
                .multilineTextAlignment(.leading)

            // Icon/Media placeholder
            Image(systemName: icon)
                .font(.system(size: 24))
                .foregroundColor(colors.primary)
                .frame(maxWidth: .infinity, alignment: .center)
                .padding(.vertical, spacing.lg)
                .background(colors.surface)
                .cornerRadius(12)

            // Actions
            HStack(spacing: spacing.lg) {
                Button(action: {
                    isLiked.toggle()
                    currentLikes += isLiked ? 1 : -1
                }) {
                    HStack(spacing: spacing.xs) {
                        Image(systemName: isLiked ? "heart.fill" : "heart")
                            .foregroundColor(isLiked ? .red : colors.textSecondary)
                        Text("\(currentLikes)")
                            .font(.system(size: 14))
                            .foregroundColor(colors.textSecondary)
                    }
                }

                Button(action: {}) {
                    HStack(spacing: spacing.xs) {
                        Image(systemName: "bubble.left")
                            .foregroundColor(colors.textSecondary)
                        Text("\(comments)")
                            .font(.system(size: 14))
                            .foregroundColor(colors.textSecondary)
                    }
                }

                Button(action: {}) {
                    Image(systemName: "square.and.arrow.up")
                        .foregroundColor(colors.textSecondary)
                }

                Spacer()
            }
        }
        .padding(spacing.md)
        .background(Color.white)
        .cornerRadius(12)
        .shadow(color: colors.shadowLight, radius: 4, y: 2)
    }
}

// MARK: - Discover View
private struct DiscoverView: View {
    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        ScrollView {
            VStack(spacing: spacing.md) {
                Text("Trending Topics")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(colors.textPrimary)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding(.horizontal, spacing.md)

                // Trending grid placeholder
                LazyVGrid(columns: [
                    GridItem(.flexible()),
                    GridItem(.flexible())
                ], spacing: spacing.sm) {
                    ForEach(0..<6) { index in
                        RoundedRectangle(cornerRadius: 12)
                            .fill(colors.surface)
                            .frame(height: 120)
                            .overlay(
                                VStack {
                                    Image(systemName: "flame.fill")
                                        .font(.system(size: 24))
                                        .foregroundColor(colors.primary)
                                    Text("Topic \(index + 1)")
                                        .font(.system(size: 14, weight: .medium))
                                        .foregroundColor(colors.textPrimary)
                                }
                            )
                    }
                }
                .padding(.horizontal, spacing.md)
            }
            .padding(.top, spacing.sm)
        }
    }
}

// MARK: - Notifications View
private struct NotificationsView: View {
    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        ScrollView {
            VStack(spacing: spacing.sm) {
                ForEach(0..<5) { index in
                    HStack(spacing: spacing.md) {
                        Circle()
                            .fill(colors.primary.opacity(0.2))
                            .frame(width: 40, height: 40)
                            .overlay(
                                Image(systemName: "bell.fill")
                                    .font(.system(size: 16))
                                    .foregroundColor(colors.primary)
                            )

                        VStack(alignment: .leading, spacing: spacing.xs) {
                            Text("Notification \(index + 1)")
                                .font(.system(size: 14, weight: .medium))
                                .foregroundColor(colors.textPrimary)

                            Text("This is a sample notification message")
                                .font(.system(size: 12))
                                .foregroundColor(colors.textSecondary)
                        }

                        Spacer()

                        Text("2m")
                            .font(.system(size: 12))
                            .foregroundColor(colors.textSecondary)
                    }
                    .padding(.horizontal, spacing.md)
                    .padding(.vertical, spacing.sm)
                }
            }
            .padding(.top, spacing.sm)
        }
    }
}

// MARK: - Camera View (Placeholder)
private struct CameraView: View {
    @Environment(\.presentationMode) var presentationMode

    private let colors = Colors()

    var body: some View {
        VStack {
            Text("Camera")
                .font(.title)
                .foregroundColor(colors.textPrimary)

            Button("Close") {
                presentationMode.wrappedValue.dismiss()
            }
            .foregroundColor(colors.primary)
        }
    }
}

// MARK: - Preview
#if DEBUG
struct SocialScreen_Previews: PreviewProvider {
    static var previews: some View {
        SocialScreen()
            .previewDisplayName("Social Screen")
    }
}
#endif